### PR TITLE
[DDW-656] Log Daedalus state snapshot

### PR DESCRIPTION
**Pull request by DeeJayElly**
This PR adds new log for current state of the app at the moment of submitting new support request.

## Todos

- [x] Add missing parameters from the diagnostic screen
- [x] Add logic to store params and create log in SupportSettingsPage, in the same way like clicking on menu item Support Request

## Screenshots

Screenshot 2019-07-10 18 49 00

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1562948382344700)
- [ ] Check if clicking on Support Request creates the log file State-snapshot.json from the main menu
- [ ] Check if clicking on Support Request creates the log file  State-snapshot.json from the Support page
- [ ] Check if State-snapshot.json has all the relevant information from the diagnostic screen

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/common/types/logging.types.js [hunk 1]
```diff
@@ -48,3 +50,44 @@ export type LogSystemInfoParams = {
   ram: string,
   startTime: string,
 };
+
+export type LogStateSnapshotParams = {
```
Comment (daniloprates) [inline source/common/types/logging.types.js]: I can see most of the props are the same as the Diagnostics component. For clarity, you could do the following:

At `components/status/DaedalusDiagnostics.js`:

```diff
- type Props = {
+ export type DaedalusDiagnosticsProps {
```

Then here:

```js
export type LogStateSnapshotParams = {
  ...$Exact<DaedalusDiagnosticsProps>,
 // add the remaining props
}
```

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 2]
```diff
@@ -145,6 +149,8 @@ export default class NetworkStatusStore extends Store {
     this._checkDiskSpace();
 
     this._getStateDirectoryPath();
+
+    this.actions.networkStatus.logStateSnapshot.listen(this._logStateSnapshot);
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @DeeJayElly please move this line to the top after this one: https://github.com/input-output-hk/daedalus/blob/6d96fd94a98ce42c03380307563b4003d6ef6fb7/source/renderer/app/stores/NetworkStatusStore.js#L112 🙏

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 3]
```diff
@@ -304,6 +310,11 @@ export default class NetworkStatusStore extends Store {
     return Promise.resolve();
   };
 
+  _handleLogStateSnapshot = (): Promise<void> => {
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @DeeJayElly please add a comment above this line with an explanation of what this method does 🙏

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 4]
```diff
@@ -612,6 +634,66 @@ export default class NetworkStatusStore extends Store {
     this.stateDirectoryPath = stateDirectoryPath;
   };
 
+  @action _logStateSnapshot = () => {
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @DeeJayElly please add a comment above this line with an explanation of what this method does 🙏 
PS. Why is this method defined as mobx action? AFAIK it does not mutate the state...

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 5]
```diff
@@ -583,6 +594,17 @@ export default class NetworkStatusStore extends Store {
     openLocalDirectoryChannel.send(path);
   }
 
+  convertBytesToSize = (bytes: number): string => {
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @DeeJayElly I feel like this method belongs to utils rather then here. Can you please check if we already have a similar one? I am pretty sure we do...

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 6]
```diff
@@ -122,6 +124,8 @@ export default class NetworkStatusStore extends Store {
     // Passively receive state changes of the cardano-node
     cardanoStateChangeChannel.onReceive(this._handleCardanoNodeStateChange);
 
+    setLogStateSnapshotChannel.onReceive(this._handleLogStateSnapshot);
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @DeeJayElly please add a comment above this line explaining what it does

File: source/renderer/app/containers/settings/categories/SupportSettingsPage.js [hunk 7]
```diff
@@ -39,7 +39,8 @@ export default class SupportSettingsPage extends Component<InjectedProps> {
   };
 
   handleDownloadLogs = () => {
-    const { app } = this.props.actions;
+    const { app, networkStatus } = this.props.actions;
+    networkStatus.logStateSnapshot.trigger();
```
Comment (nikolaglumac) [inline source/renderer/app/containers/settings/categories/SupportSettingsPage.js]: @DeeJayElly are we 100% what this action does will finish before the packing/download log starts?

**Comment by nikolaglumac** [comment]
@tomothespian please review this one too 🙏

**Comment by gnpf** [comment]
@DeeJayElly The value of _isNodeSyncing_ is incorrect on the file.

image

**Comment by gnpf** [comment]
> @DeeJayElly The value of _isNodeSyncing_ is incorrect on the file.
> 
> image

@DeeJayElly Please ignore this! The file was created while still connecting to network, so the sync hadn't started yet.

File: source/common/types/logging.types.js (hunk 1 above)
Comment (DeeJayElly) [reply source/common/types/logging.types.js]: Done

File: source/renderer/app/stores/NetworkStatusStore.js (hunk 2 above)
Comment (DeeJayElly) [reply source/renderer/app/stores/NetworkStatusStore.js]: Done

File: source/renderer/app/stores/NetworkStatusStore.js (hunk 3 above)
Comment (DeeJayElly) [reply source/renderer/app/stores/NetworkStatusStore.js]: Done

File: source/renderer/app/stores/NetworkStatusStore.js (hunk 4 above)
Comment (DeeJayElly) [reply source/renderer/app/stores/NetworkStatusStore.js]: Done

File: source/renderer/app/stores/NetworkStatusStore.js (hunk 5 above)
Comment (DeeJayElly) [reply source/renderer/app/stores/NetworkStatusStore.js]: Done

File: source/renderer/app/stores/NetworkStatusStore.js (hunk 6 above)
Comment (DeeJayElly) [reply source/renderer/app/stores/NetworkStatusStore.js]: Done

File: source/renderer/app/containers/settings/categories/SupportSettingsPage.js (hunk 7 above)
Comment (DeeJayElly) [reply source/renderer/app/containers/settings/categories/SupportSettingsPage.js]: Done

File: source/common/types/logging.types.js (hunk 1 above)
Comment (nikolaglumac) [reply source/common/types/logging.types.js]: This was not a good idea as the props are not equal.
Please revert this @DeeJayElly 
cc @daniloprates

**Comment by gnpf** [comment]
@DeeJayElly 
The Daedalus version is displayed twice.

image

**Comment by nikolaglumac** [comment]
@gnpf please focus on the stuff within the `data` attribute. Everything else is system output which has to be like that (there is some duplication).

**Pull request merged**